packages/core/src/utils/parseInvitation.ts

```diff
@@ -1,10 +1,122 @@
+import type { AgentDependencies } from '../agent/AgentDependencies'
+import type { Response } from 'node-fetch'
+
 import { parseUrl } from 'query-string'
 
 import { AriesFrameworkError } from '../error'
 import { ConnectionInvitationMessage } from '../modules/connections'
 import { convertToNewInvitation } from '../modules/oob/helpers'
 import { OutOfBandInvitation } from '../modules/oob/messages'
 
+import { JsonEncoder } from './JsonEncoder'
+import { JsonTransformer } from './JsonTransformer'
+import { MessageValidator } from './MessageValidator'
+
+const fetchShortUrl = async (invitationUrl: string, dependencies: AgentDependencies) => {
+  // eslint-disable-next-line no-restricted-globals
+  const abortController = new AbortController()
+  const id = setTimeout(() => abortController.abort(), 15000)
+  let response
+  try {
+    response = await dependencies.fetch(invitationUrl, {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+    })
+  } catch (error) {
+    throw new AriesFrameworkError('Get request failed on provided Url')
```

Comment:
We'll lose all information on why this failed here. Maybe we can include some of the error details here:

```suggestion
    throw new AriesFrameworkError(`Get request failed on provided url: ${error.message}`, { cause: error })
```